File: frontend/src/app/service/http/model/range.ts
import {RangeOperator} from './range-operator.enum';

export interface RangeInterface {
  operator: RangeOperator;
  property: string;
  value1?: any;
  value2?: any;
  values?: any[];
}

export class Range {
  exclude: boolean = false;
  operator: RangeOperator;
  property: string;
  value1?: any;
  value2?: any;
  values?: any[];

  constructor(params: RangeInterface) {
    this.operator = params.operator;
    this.property = params.property;
    this.value1 = params.value1;
    this.value2 = params.value2;
    this.values = params.values;
  }
}

File: backend/src/main/java/com/webvrlamov/alstromeria/core/filter/models/Range.java
package com.webvrlamov.alstromeria.core.filter.models;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.webvrlamov.alstromeria.core.filter.service.CustomRangeValueDeserializer;
import lombok.Data;

import java.util.List;

@Data
public class Range {
    @Deprecated
    boolean exclude;
    RangeOperator operator;
    String property;
    @JsonDeserialize(using = CustomRangeValueDeserializer.class)
    Object value1;
    @JsonDeserialize(using = CustomRangeValueDeserializer.class)
    Object value2;
    List<Object> values;

    public Range() {}

    public Range(boolean exclude, RangeOperator operator, String property) {
        this.exclude = exclude;
        this.operator = operator;
        this.property = property;
    }

    public Range setExclude(boolean exclude) {
        this.exclude = exclude;
        return this;
    }

    public Range setOperator(RangeOperator operator) {
        this.operator = operator;
        return this;
    }

    public Range setProperty(String property) {
        this.property = property;
        return this;
    }

    public Range setValue1(Object value1) {
        this.value1 = value1;
        return this;
    }

    public Range setValue2(Object value2) {
        this.value2 = value2;
        return this;
    }

    public Range setValues(List<Object> values) {
        this.values = values;
        return this;
    }
}
